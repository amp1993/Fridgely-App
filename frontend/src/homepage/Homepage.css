/* Homepage.css */

.homepage-container {
    display: flex;
    justify-content: center; /* Center horizontally */
    align-items: center;     /* Center vertically */
    height: 100vh;           /* Full viewport height */
    text-align: center;      /* Center text within container */
}

.homepage-content {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.homepage-top-box {
    margin-bottom: 20px;    
}

.homepage-links {
    display: flex;           
    gap: 10px;               
}

.homepage-link-box {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 150px;            
    height: 150px;           
    margin: 10px;            
    text-align: center;      
    padding: 10px;
    border: 1px solid #3498db; 
    border-radius: 10px;     
    color: #3498db;
    cursor: pointer;
    text-decoration: none;
}

.homepage-link-box:hover {
    background-color: #3498db;
    color: #fff;
}
