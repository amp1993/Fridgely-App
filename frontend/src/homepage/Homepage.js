import React from "react";
import { Card, CardBody, CardTitle } from "reactstrap";
import { NavLink, Link } from "react-router-dom";
import Profile from "../user/Profile";

import "./Homepage.css";

const Homepage = ({ user }) => {
    return (
        <div className="homepage-container">
            {user ? (
           // If the user is logged in, display the Profile component

                <>

                    <Profile user={user} />
                </>
            ) : (
            // If the user is not logged in, display the homepage content

                <>
                    <section className="homepage-content">
                        <Card className="homepage-top-box">
                            <CardBody className="text-center">
                                <CardTitle className="font-weight-bold">
                                    <h1 className="page-title">
                                        Manage Your Meals With <strong id="logo">Fridgely</strong>
                                    </h1>
                                    <p className="tagline">
                                        Effortless Meal Management: Simplify Your Kitchen with Fridgely
                                    </p>
                                </CardTitle>
                            </CardBody>
                        </Card>
                        <div className="homepage-links">
                            <Link to="/register" className="homepage-link-box">
                                Create an Account
                            </Link>
                            <Link to="/login" className="homepage-link-box">
                                Log In
                            </Link>
                            <Link to="/recipes" className="homepage-link-box">
                                Browse Recipes
                            </Link>
                        </div>
                    </section>
                </>
            )}
        </div>
    );
};

export default Homepage;
