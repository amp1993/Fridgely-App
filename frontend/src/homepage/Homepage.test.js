import React from 'react';
import { render, within } from '@testing-library/react';
import FridgelyApi from '../api/api';
import { BrowserRouter } from 'react-router-dom';

import Homepage from "./Homepage"

it("renders without crashing", function() {
    render(
    <BrowserRouter>
        <Homepage />
      </BrowserRouter>);
  });

  it("matches snapshot", function() {
    const { asFragment } = render(
    <BrowserRouter>
        <Homepage />
      </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
  });


  it('displays the correct texts for logged-out users', function() {
    const { getByText } = render(
        <BrowserRouter>
            <Homepage />
        </BrowserRouter>
    );

    const cardTitle = getByText('Manage Your Meals With', { exact: false }).closest('h1');
    expect(cardTitle).toBeInTheDocument();
    expect(within(cardTitle).getByText('Fridgely')).toBeInTheDocument();

    expect(getByText('Effortless Meal Management: Simplify Your Kitchen with Fridgely')).toBeInTheDocument();
    expect(getByText('Create an Account')).toBeInTheDocument();
    expect(getByText('Log In')).toBeInTheDocument();
    expect(getByText('Browse Recipes')).toBeInTheDocument();
});
