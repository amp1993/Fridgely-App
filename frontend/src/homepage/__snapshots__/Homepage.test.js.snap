// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="homepage-container"
  >
    <section
      class="homepage-content"
    >
      <div
        class="homepage-top-box card"
      >
        <div
          class="text-center card-body"
        >
          <div
            class="font-weight-bold card-title"
          >
            <h1
              class="page-title"
            >
              Manage Your Meals With 
              <strong
                id="logo"
              >
                Fridgely
              </strong>
            </h1>
            <p
              class="tagline"
            >
              Effortless Meal Management: Simplify Your Kitchen with Fridgely
            </p>
          </div>
        </div>
      </div>
      <div
        class="homepage-links"
      >
        <a
          class="homepage-link-box"
          href="/register"
        >
          Create an Account
        </a>
        <a
          class="homepage-link-box"
          href="/login"
        >
          Log In
        </a>
        <a
          class="homepage-link-box"
          href="/recipes"
        >
          Browse Recipes
        </a>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
